### ISSUE TITLE
Cinnamon Maximus (forked by Odyseus) - Transparent windows (Cinnamon 3.4)

### ISSUE THREAD

**Issue by natanvarga** [Odyseus/CinnamonTools#84]
With the recent Cinnamon 3.4 update, maximised windows are undecorated as they should be, but some of them also become transparent, i.e., none of the window content can be seen. Such applications are vivaldi, chromium, cinnamon settings, mousepad, libreoffice-writer, etc. However, some other apps don't exhibit the same behaviour, e.g., xfce4-terminal does not become transparent, the same is true for nemo. There doesn't seem to be a pattern yet, but most apps I've tried do become transparent upon maximising. Upon Alt+Tab I can see the undecorated and maximised window.

**Comment by Odyseus** [comment]
Hello, @natanvarga.

See if any of what I said [here](https://github.com/Odyseus/CinnamonTools/issues/55#issuecomment-299619433) helps. It refers to a tweak in the **Cinnamon Tweaks** extension, but it applies to the Maximus extension as well because its the exact same code. See if the alternative option suits your needs too.

**Comment by natanvarga** [comment]
Okay. Thanks

**Comment by Odyseus** [comment]
Hello, @natanvarga.

I tracked down the problem of invisible windows and fixed it (at least, I cannot reproduce it anymore on Cinnamon 3.4.x). If you are still using the extension, with the next update (I published it just now and awaiting review) there will be a new option called **Enable hack for invisible windows**. After enabling this option, all windows with decorations removed needs to be restarted. Read the option tooltip for know *side effects*.
